lib/statsd/class-statsd.php

```diff
@@ -0,0 +1,146 @@
+	 *
+	 * @param string|array $stats The metric(s) to set.
+	 * @param float $time The elapsed time (ms) to log
+	 **/
+	public static function timing( $stats, $time ) {
+		self::update_stats( $stats, $time, 1, 'ms' );
+	}
+
+	/**
+	 * Sets one or more gauges to a value
+	 *
+	 * @param string|array $stats The metric(s) to set.
+	 * @param float $value The value for the stats.
+	 **/
+	public static function gauge( $stats, $value ) {
+		self::update_stats( $stats, $value, 1, 'g' );
+	}
+
+	/**
+	 * A "Set" is a count of unique events.
+	 * This data type acts like a counter, but supports counting
+	 * of unique occurences of values between flushes. The backend
+	 * receives the number of unique events that happened since
+	 * the last flush.
+	 *
+	 * The reference use case involved tracking the number of active
+	 * and logged in users by sending the current userId of a user
+	 * with each request with a key of "uniques" (or similar).
+	 *
+	 * @param string|array $stats The metric(s) to set.
+	 * @param float $value The value for the stats.
+	 **/
+	public static function set( $stats, $value ) {
+		self::update_stats( $stats, $value, 1, 's' );
+	}
+
+	/**
+	 * Increments one or more stats counters
+	 *
+	 * @param string|array $stats The metric(s) to increment.
+	 * @param float|1 $sample_rate the rate (0-1) for sampling.
+	 * @return boolean
+	 **/
+	public static function increment( $stats, $sample_rate = 1 ) {
+		self::update_stats( $stats, 1, $sample_rate, 'c' );
+	}
+
+	/**
+	 * Decrements one or more stats counters.
+	 *
+	 * @param string|array $stats The metric(s) to decrement.
+	 * @param float|1 $sample_rate the rate (0-1) for sampling.
+	 * @return boolean
+	 **/
+	public static function decrement( $stats, $sample_rate = 1 ) {
+		self::update_stats( $stats, -1, $sample_rate, 'c' );
+	}
+
+	/**
+	 * Updates one or more stats.
+	 *
+	 * @param string|array $stats The metric(s) to update. Should be either a string or array of metrics.
+	 * @param int|1 $delta The amount to increment/decrement each metric by.
+	 * @param float|1 $sample_rate the rate (0-1) for sampling.
+	 * @param string|c $metric The metric type ("c" for count, "ms" for timing, "g" for gauge, "s" for set)
+	 * @return boolean
+	 **/
+	public static function update_stats( $stats, $delta = 1, $sample_rate = 1, $metric = 'c' ) {
+		if ( ! is_array( $stats ) ) {
+			$stats = array( $stats ); 
+		}
+
+		$data = array();
+
+		foreach ( $stats as $stat ) {
+			$data[ $stat ] = "$delta|$metric";
+		}
+
+		self::send( $data, $sample_rate );
+	}
+
+	/*
+	 * Send the metrics over UDP
+	 **/
+	public static function send( $data, $sample_rate = 1 ) {
+		// Disables StatsD logging for test environments
+		if ( defined( 'VIP_DISABLE_STATSD' ) && VIP_DISABLE_STATSD ) {
+			return;
+		}
+
+		// If we don't have server info defined, abort
+		if ( ! defined( 'VIP_STATSD_HOST' ) || ! VIP_STATSD_HOST || ! defined( 'VIP_STATSD_PORT' ) || ! VIP_STATSD_PORT ) {
+			return;
```

Comment:
In light of [this thread](https://github.com/Automattic/vip-go-mu-plugins/pull/1521#discussion_r400392704) above, probably not, since this will fire in local environments that aren't setup for StatsD.